packages/aws_common/lib/src/util/uuid.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:uuid/uuid.dart';
+import 'package:uuid/uuid_util.dart';
+
+/// Generates a crypto-secure UUID (v4).
+String uuid() => const Uuid().v4(
```

Comment:
Is the expectation that all current uses of Uuid().v4 are replaced with this? Is there a downside to using this in locations where a non crypto secure uuid would be sufficient?

Reply:
> Is the expectation that all current uses of Uuid().v4 are replaced with this?
Yes, at least in Smithy/AWS we'll have to. I think we should also replace it throughout Amplify.

> Is there a downside to using this in locations where a non crypto secure uuid would be sufficient?
Maybe a slight performance hit. The converse, though, is that there is a security risk if we use it incorrectly.